app/api/facility/[id]/route.ts

```diff
@@ -0,0 +1,72 @@
+import { deleteFile, uploadFile } from "@/app/providers/S3Provider";
+import { facilityProvider } from "@/app/providers/facilityProvider";
+import { FacilityProfile } from "@prisma/client";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function POST(request: NextRequest) {
+  try {
+    const urlArr = request.url.split("/");
+    const userId = urlArr[urlArr.length - 1];
```

Comment:
I believe you should be able to do
```suggestion
    const { id } = req.query;;
```